mne/io/eeglab/eeglab.py

```diff
@@ -0,0 +1,468 @@
+                    # store latency of only first event
+                    event_latencies.append(eeg.event[ev_idx].latency)
+                    ev_idx += len(ep.eventtype)
+                    warnings.warn('An epoch has multiple events. '
+                                  'Only the latency of first event will be '
+                                  'retained.')
+                else:
+                    event_type = ep.eventtype
+                    event_name.append(ep.eventtype)
+                    event_latencies.append(eeg.event[ev_idx].latency)
+                    ev_idx += 1
+
+                if event_type not in unique_ev:
+                    unique_ev.append(event_type)
+
+                # invent event dict but use id > 0 so you know its a trigger
+                event_id = dict((ev, idx + 1) for idx, ev
+                                in enumerate(unique_ev))
+            # now fill up the event array
+            events = np.zeros((eeg.trials, 3), dtype=int)
+            for idx in range(0, eeg.trials):
+                if idx == 0:
+                    prev_stim = 0
+                elif (idx > 0 and
+                        event_latencies[idx] - event_latencies[idx - 1] == 1):
+                    prev_stim = event_id[event_name[idx - 1]]
+                events[idx, 0] = event_latencies[idx]
+                events[idx, 1] = prev_stim
+                events[idx, 2] = event_id[event_name[idx]]
+        elif isinstance(events, string_types):
+            events = read_events(events)
+
+        logger.info('Extracting parameters from %s...' % input_fname)
+        input_fname = op.abspath(input_fname)
+        info = _get_info(eeg, montage, eog=eog)
+
+        for key, val in event_id.items():
+            if val not in events[:, 2]:
+                raise ValueError('No matching events found for %s '
+                                 '(event id %i)' % (key, val))
+
+        self._filename = input_fname
+        if isinstance(eeg.data, string_types):
+            basedir = op.dirname(input_fname)
+            data_fname = op.join(basedir, eeg.data)
+            _check_fname(data_fname)
+            with open(data_fname, 'rb') as data_fid:
+                data = np.fromfile(data_fid, dtype=np.float32)
+                data = data.reshape((eeg.nbchan, eeg.pnts, eeg.trials),
+                                    order="F")
+        else:
+            data = eeg.data
+        data = data.transpose((2, 0, 1)).astype('double')
+        _rescale_data(info, data)
+        assert data.shape == (eeg.trials, eeg.nbchan, eeg.pnts)
+        tmin, tmax = eeg.xmin, eeg.xmax
+
+        super(EpochsEEGLAB, self).__init__(
+            info, data, events, event_id, tmin, tmax, baseline,
+            reject=reject, flat=flat, reject_tmin=reject_tmin,
+            reject_tmax=reject_tmax, add_eeg_ref=False, verbose=verbose)
+
+        # XXX: hack because add_eeg_ref in _BaseEpochs needs to be fixed. Also
```

Comment:
It doesn't work just to have `add_eeg_ref=add_eeg_ref` above? We need to fix it for release. :(